### PR TITLE
[docs-infra] Transpile .ts demo files

### LINKED ISSUE #15059: [docs] Chart example with responsive container and tooltip demo is broken
### Related page

https://mui.com/x/react-charts/tooltip/#tooltip-position

### Kind of issue

Broken demo

### Issue description

Opening example in code editor displays missing files

image

### Context

_No response_

**Search keywords**: chart, charts, responsive, tooltip

### PR DESCRIPTION
**Pull request by KenanYusuf**
Fix #15059

- Update `formattedTSDemos.js` script to transpile `.ts` files
- Run transpilation script to get `.js` equivalents of `.ts` files in existing demos

Related to https://github.com/mui/material-ui/pull/44150

### CODE REVIEW

File: docs/scripts/formattedTSDemos.js [hunk 1]
```diff
@@ -79,7 +81,7 @@ const TranspileResult = {
 };
 
 async function transpileFile(tsxPath, program, ignoreCache = false) {
-  const jsPath = tsxPath.replace('.tsx', '.js');
+  const jsPath = tsxPath.replace(/\.tsx?$/, '.js');
```
Comment (KenanYusuf) [inline docs/scripts/formattedTSDemos.js:84]: To include `.ts` files

File: docs/scripts/formattedTSDemos.js [hunk 2]
```diff
@@ -42,7 +42,7 @@ const babelConfig = {
 
 const workspaceRoot = path.join(__dirname, '../../');
 
-async function getFiles(root) {
+async function getFiles(root, excludeRoot = false) {
```
Comment (KenanYusuf) [inline docs/scripts/formattedTSDemos.js:45]: I don't think we want to transpile files at the root of `docs/data`, e.g. https://github.com/mui/mui-x/blob/master/docs/data/chartsApiPages.ts

**Review by alexfauquette — APPROVED** [review]
Thanks for handling that :+1:

Did you checked if it works when updating the `@mui-internals/markdown`?

**Comment by KenanYusuf** [comment]
@alexfauquette is there something in particular that needs testing with that package? If it's related to https://github.com/mui/material-ui/pull/44150, we're on `1.0.19` which includes those changes

**Comment by alexfauquette** [comment]
> we're on 1.0.19 which includes those changes

I missed this update. Then it's all good, I tried the docs preview and nested files work perfectly on charts demo 🚀👌

**Pull request merged**